ee/hogai/taxonomy_agent/nodes.py

```diff
@@ -226,7 +226,12 @@ def _construct_messages(self, state: AssistantState) -> list[BaseMessage]:
                     conversation.append(LangchainHumanMessage(content=message.content))
             elif isinstance(message, VisualizationMessage):
                 conversation.append(LangchainAssistantMessage(content=message.plan or ""))
-            elif isinstance(message, AssistantMessage):
+            elif isinstance(message, AssistantMessage) and (
+                idx < 1
+                or not isinstance(
+                    filtered_messages[idx - 1], VisualizationMessage
+                )  # Filter out the summarizer messages.
```

Comment:
Found this a bit hard to parse:
```suggestion
                # Filter out summarizer messages (which always follow viz), but leave clarification questions in
                idx < 1
                or not isinstance(
                    filtered_messages[idx - 1], VisualizationMessage
                )
```